ui/helpers/countdown.ts

```diff
@@ -1,3 +1,11 @@
+/** @desc Creating Intl.NumberFormat is relatively slow, therefore using memoize() per set of arguments */
+const createNumberFormat = _.memoize(
+  (...[locale, options]: Parameters<typeof Intl.NumberFormat>) =>
+    new Intl.NumberFormat(locale, options),
+  (...[locale, options]: Parameters<typeof Intl.NumberFormat>) =>
+    [locale, options?.unit, options?.maximumFractionDigits].join("|"),
+);
+
 export const formatDeadline = (
   time: number,
   locales = [LOCALE, undefined],
```

Comment:
it can also be having underscore `_` for some reason.
intermediate fallback with replacing it to `-` could be beneficial.

Reply:
addressed